### PR TITLE
Generate openvex data on each release

### PR DESCRIPTION
**Pull request by psschwei**
Signed-off-by: Paul S. Schweigert <paul@paulschweigert.com>

`#SecuritySlam`

#### What this PR does / why we need it:

This PR adds a github action to generate openvex data on each release.

#### Which issue(s) this PR fixes:

This commit is part of the December 2023 Security Slam.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/assign @ArangoGutierrez

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
For context @alculquicondor -> https://docs.google.com/document/d/1TLahqW6OpWQXbaVJ-SR8z-o0yLSoGKYTHxPU0PyyjXo/edit

**Review by ArangoGutierrez — APPROVED** [review]
This RP is part of the SecuritySlam organised by CNCF
@alculquicondor @tenzen-y

File: .github/workflows/openvex.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - uses: openvex/generate-vex@31b415924ea0d72ed5f2640f1dee59dea6c2770b
+        name: Run vexctl
+        with:
+            product: pkg:generic/hello@${{ env.RELEASE_VERSION }}
```
Comment (alculquicondor) [inline .github/workflows/openvex.yaml:17]: is this a placeholder?

**Review by alculquicondor — COMMENTED** [review]
/approve
Leaving LGTM to @ArangoGutierrez

File: .github/workflows/openvex.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - uses: openvex/generate-vex@31b415924ea0d72ed5f2640f1dee59dea6c2770b
+        name: Run vexctl
+        with:
+            product: pkg:golang/sigs.k8s.io/kueue@${{ env.RELEASE_VERSION }}
```
Comment (psschwei) [inline .github/workflows/openvex.yaml:17]: @puerco can you verify that the purl here is correct?

File: .github/workflows/openvex.yaml (hunk 1 above)
Comment (psschwei) [reply .github/workflows/openvex.yaml]: Good catch. Yes it was, I've updated it.

File: .github/workflows/openvex.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply .github/workflows/openvex.yaml:17]: the previous step set's the envVar 
```yaml
      - name: Set env
        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
```

Comment (ArangoGutierrez) [reply .github/workflows/openvex.yaml:17]: So it is OK

Comment (shafeeqes) [reply .github/workflows/openvex.yaml:17]: @psschwei Hey, How did you derive at this purl?

Comment (psschwei) [reply .github/workflows/openvex.yaml:17]: from the purl-spec repo: https://github.com/package-url/purl-spec

Comment (shafeeqes) [reply .github/workflows/openvex.yaml:17]: Thanks!

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**